providers/hetzner/api.go

```diff
@@ -236,23 +281,55 @@ func (requestRateLimiter *requestRateLimiter) beforeRequest() {
 	time.Sleep(time.Until(requestRateLimiter.lastRequest.Add(requestRateLimiter.delay)))
 }
 
-func (requestRateLimiter *requestRateLimiter) bumpDelay() string {
-	var backoffType string
-	if requestRateLimiter.delay == 0 {
-		// At the time this provider was implemented (2020-10-18),
-		//  one request per second could go though when rate-limited.
-		requestRateLimiter.delay = time.Second
-		backoffType = "constant"
-	} else {
-		// The initial assumption of 1 req/s may no hold true forever.
-		// Future proof this provider, use exponential back-off.
-		requestRateLimiter.delay = requestRateLimiter.delay * 2
-		backoffType = "exponential"
+func (requestRateLimiter *requestRateLimiter) setDefaultDelay() {
+	// default to a rate-limit of 1 req/s -- the next response should update it.
+	requestRateLimiter.delay = time.Second
+}
+
+func (requestRateLimiter *requestRateLimiter) setOptimizeForRateLimitQuota(quota string) error {
+	quotaNormalized := strings.ToLower(quota)
```

Comment:
Suggestion: The code would be more simple if you used ToLower'ed strings.  In other words, if you used `hour` instead of `Hour`, the case statement below could be replaced with just:

    requestRateLimiter.optimizeForRateLimitQuota = quoteNormalized

Reply:
Thanks for the suggestion. I also found `strings.Title()` for casting it back when reading the header -- Hetzner is using the header name `X-Ratelimit-Limit-Hour`. I am going to retain the switch logic as I think it's a nice way of validating the user provided value and providing a fallback for the empty string. da730e103b05dd749104911627e4834928757777